### DISCUSSION TITLE
Getting this error: 'BigUint64Array' not defined under the 'no-undef' lint

### DISCUSSION THREAD

**Discussion by akshay1992kalbhor** [eslint/eslint#14663] [Q&A]
Why can't I use BigUint64Array in my project? I can use it as a type annotation for function arguments but construct a value inside a function using the new keyword.

**Comment by aladdin-add** [comment]
can you post your eslint version & config please?

**Comment by tamish-max** [comment]
`BigUint64Array` is an ES2020 global. If ESLint is reporting it under `no-undef`, your config is probably using an older ECMAScript environment.

Something like this should fix it:

```js
module.exports = {
  env: { es2020: true },
  parserOptions: { ecmaVersion: 2020 },
}
```

Or you can declare it explicitly in `globals`, but usually setting the right `env` / `ecmaVersion` is the better fix.

**Comment by barry166** [comment]
`BigUint64Array` is an ES2020 global. TypeScript can know it as a type, but ESLint's `no-undef` checks runtime globals separately, so `new BigUint64Array()` is reported unless your ESLint config includes ES2020 globals.

With flat config, add the ES2020 globals:

```js
import globals from "globals"

export default [
  {
    languageOptions: {
      ecmaVersion: 2020,
      globals: globals.es2020,
    },
  },
]
```

In older `.eslintrc`, the equivalent is `env: { es2020: true }`. If the runtime you target does not actually support `BigUint64Array`, then the lint error is correct and you should not add the global.
